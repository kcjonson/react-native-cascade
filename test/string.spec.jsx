import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import StringComponent from './StringComponent';


/*

  These tests check the usage of <* className="*"> as a string property

*/


it('matches the snapshot for a string className attribute', () => {
  const tree = renderer
    .create(<StringComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('creates a style node with appropriate style for a string className attribute', () => {
  const com = shallow(<StringComponent />);
  const hasStylesNode = com.find({'test-id': 'base'});
  expect(JSON.stringify(hasStylesNode.prop('style'))).toBe(JSON.stringify({
    color: 'red',
    fontSize: 24,
    backgroundColor: 'chartreuse',
    paddingTop: 16,
  }));
});

it('matches mutliple classname selectors', () => {
  const com = shallow(<StringComponent />);
  const hasStylesNode = com.find({'test-id': 'mutliple-selector'});
  expect(JSON.stringify(hasStylesNode.prop('style'))).toBe(JSON.stringify({
    color: 'green',
    fontSize: 32,
    backgroundColor: 'chartreuse',
  }));
});


it('will add (merge) to an existing style definition for a string className attribute', () => {
  const com = shallow(<StringComponent />);
  ['style-object', 'style-object-reverse'].forEach(testId => {
    const hasStylesNode = com.find({'test-id': testId});
    expect(JSON.stringify(hasStylesNode.prop('style'))).toBe(JSON.stringify([
      {
        color: 'red',
        fontSize: 24,
        backgroundColor: 'chartreuse',
        paddingTop: 16,
      }, {
        color: 'pink',
      },
    ]));
  });
});

it('will add (merge) to an existing style definition that is an array for a string className attribute', () => {
  const com = shallow(<StringComponent />);
  const hasStylesNode = com.find({'test-id': 'style-unmatched'});
  expect(JSON.stringify(hasStylesNode.prop('style'))).toBe('[{"paddingTop":16},{"backgroundColor":"pink"}]');
});

// These are tests for build time optimizations that are currently turned off.

// it("doesn't add to style tags for string className attributes with unmatched selectors", () => {
//   const com = shallow(<StringComponent />);
//   const hasStylesNode = com.find({'test-id': 'unmatched'});
//   expect(hasStylesNode.prop('style')).toBeUndefined();
// });
//
// it("doesn't create style tags for string className attributes with unmatched selectors", () => {
//   const com = shallow(<StringComponent />);
//   const hasStylesNode = com.find({'test-id': 'unmatched'});
//   expect(hasStylesNode.prop('style')).toBeUndefined();
// });

// it('doesnt have the runtime when not needed', () => {
//   const com = shallow(<StringComponent />);
//   expect(() => {
//     com.instance().getComputestyleRuntime();
//   }).toThrowError();
// });

// TODO: expect a warning at build time if a style is undec
