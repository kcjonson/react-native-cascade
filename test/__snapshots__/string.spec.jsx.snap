// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot for a string className attribute 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    className="text-large container"
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "chartreuse",
        "color": "red",
        "fontSize": 24,
        "paddingTop": 16,
      }
    }
    test-id="base"
  >
    Hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    className="container primary"
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "chartreuse",
        "color": "green",
        "fontSize": 32,
      }
    }
    test-id="mutliple-selector"
  >
    Hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    className="text-large container"
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "chartreuse",
          "color": "red",
          "fontSize": 24,
          "paddingTop": 16,
        },
        Object {
          "color": "pink",
        },
      ]
    }
    test-id="style-object"
  >
    Hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    className="text-large container"
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "chartreuse",
          "color": "red",
          "fontSize": 24,
          "paddingTop": 16,
        },
        Object {
          "color": "pink",
        },
      ]
    }
    test-id="style-object-reverse"
  >
    Hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    className="text-large container"
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "chartreuse",
          "color": "red",
          "fontSize": 24,
          "paddingTop": 16,
        },
        Array [
          Object {
            "color": "pink",
          },
        ],
      ]
    }
    test-id="style-array"
  >
    Hello
  </Text>
  <View
    className="text-medium"
    style={
      Array [
        Object {
          "paddingTop": 16,
        },
        Object {
          "backgroundColor": "pink",
        },
      ]
    }
    test-id="style-unmatched"
  />
  <View
    className="text-medium"
    style={
      Object {
        "paddingTop": 16,
      }
    }
    test-id="unmatched"
  />
</View>
`;
